server/metrics/src/lib.rs

```diff
@@ -0,0 +1,106 @@
+use std::error::Error;
+use std::fmt;
+use std::net::{Ipv4Addr, SocketAddrV4};
+use std::sync::Arc;
+
+use hyper;
+use hyper::service::service_fn_ok;
+use hyper::{Body, Response, Server};
+use prometheus::{Encoder, Registry, TextEncoder};
+
+use graph::prelude::{MetricsServer as MetricsServerTrait, *};
+
+/// Errors that may occur when starting the server.
+#[derive(Debug)]
+pub enum PrometheusMetricsServeError {
+    BindError(hyper::Error),
+}
+
+impl Error for PrometheusMetricsServeError {
+    fn description(&self) -> &str {
+        "Failed to start the server"
+    }
+
+    fn cause(&self) -> Option<&dyn Error> {
+        None
+    }
+}
+
+impl fmt::Display for PrometheusMetricsServeError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            PrometheusMetricsServeError::BindError(e) => {
+                write!(f, "Failed to bind index node server: {}", e)
+            }
+        }
+    }
+}
+
+impl From<hyper::Error> for PrometheusMetricsServeError {
+    fn from(err: hyper::Error) -> Self {
+        PrometheusMetricsServeError::BindError(err)
+    }
+}
+
+pub struct PrometheusMetricsServer {
+    logger: Logger,
+    registry: Arc<Registry>,
+}
+
+impl std::clone::Clone for PrometheusMetricsServer {
```

Comment:
I don't think you need to import `std::clone::Clone`, just `Clone` could be enough. Either way, I'd just have `Clone` here and if we need to import anything, I'd do that at the top. Same with `fmt::Display` above.